workspaces/ballerina/inline-data-mapper/src/components/Diagram/Node/PrimitiveOutput/PrimitiveOutputElementWidget.tsx

```diff
@@ -74,14 +75,16 @@ export function PrimitiveOutputElementWidget(props: PrimitiveOutputElementWidget
     let portName = getSanitizedId(parentId);
     if (fieldIndex !== undefined) {
         portName = `${portName}.${fieldIndex}`;
-    } else if (fieldName) {
-        portName = `${portName}.${fieldName}`;
```

Comment:
Have you tested this with a primitive type sub mapping? Looks like this should break it.